pages/docs/tutorials/native/interop-with-c.md

```diff
@@ -0,0 +1,140 @@
+for pretty much anything we could need. 
+
+In this tutorial we'll see how to create a native Kotlin application that uses the `libcurl` C library to call an HTTP endpoint. We'll cover
+how to 
+
+* [Create Kotlin Bindings](#generating-bindings)
+* [Consume a generated Kotlin API](#consuming-the-kotlin-api)
+* [Link the library into the application](#compiling-and-linking-the-library)
+
+
+## Generating Bindings
+
+An ideal scenario for interop is to call C functions as if we were calling Kotlin functions, that is, following the same signature and conventions. And that is precisely what the 
+`cinterop` tool provides us with. It takes a C library and generates the corresponding Kotlin bindings for it, which then allow us
+to use the library as if it were Kotlin code. 
+
+In order to generate these bindings, we need to create a library definition file (`def`) that contains some information about the headers we need to generate. In our case we want to use the famous `libcurl` library
+to make some HTTP calls, so we'll create a file named `libcurl.def` with the following contents
+
+```
+headers = curl/curl.h
+headerFilter = curl/*
+linkerOpts.osx = -L/opt/local/lib -L/usr/local/opt/curl/lib -lcurl
+linkerOpts.linux = -L/usr/lib/x86_64-linux-gnu -lcurl```
+```
+
+A few things are going on in that file so let's see them one by one. The first entry is `headers` which is the list of header files that we want to generate 
+Kotlin stubs for. We can add multiple files to this entry, separating each one with `\` on a new line. In our case we only want `curl.h`. The files we reference
+need to be relative to the folder where the definition file is, or be available on the system path (in our case it would be `/usr/bin/curl`).
+
+The second line is the `headerFilter`. This is used to denote what exactly we want included. In C, when one file references another file with the `#include` directive, 
+all the headers are also included. Sometimes this may not be needed, and we can use this parameter, [using glob patterns](https://en.wikipedia.org/wiki/Glob_(programming)), to fine tune things. 
+
+The third and forth lines are about providing linker options, which can vary based on different target platforms. In our cause we are defining it for macOS and Linux. 
+Compiler options can also be specific, albeit in this case it is not necessary. 
+
+The convention that is followed is that each library gets its own definition file, usually named the same as the library. For more information on all
+the options available to `cinterop`, see [README available on GitHub](https://github.com/JetBrains/kotlin-native/blob/master/INTEROP.md)
+
+Once we have the definition file ready, we can invoke `cinterop` and have it generate the bindings for us
+
+    cinterop -def libcurl.def -o build/c_interop/libcurl
+    
+The output should be a Kotlin compiled library (extension `.klib`) and the Kotlin source header files. We're also instructing `cinterop`
```

Comment:
Producing Kotlin source header files is not smth we want to advertise. Also it will likely go away soon (will generate descriptors instead). I'd suggest to rewrite it without mentioning source code. `klib` utility (klib contents) is the recommended way of inspecting contents of the klib.